components/formats-gpl/src/loci/formats/in/MetamorphReader.java

```diff
@@ -352,7 +351,7 @@ protected void initFile(String id) throws FormatException, IOException {
       LOGGER.info("Initializing " + id);
       // find an associated STK file
       String stkFile = id.substring(0, id.lastIndexOf("."));
-      if (stkFile.indexOf(File.separator) != -1) {
+      if (stkFile.indexOf(File.separatorChar) != -1) {
```

Comment:
Standard libraries simply have,
```java
public static final String separator = "" + separatorChar;
```
They're the same, this is really just about switching to calling a method that can assume searching for a single character instead of having to care about search string length.